management-interface.go

```diff
@@ -141,6 +141,65 @@ func IndexHandler(w http.ResponseWriter, r *http.Request) {
 	tmpl.ExecuteTemplate(w, "index.html", nil)
 }
 
+// Type used in serving interface information.
+type interfaceDetails struct {
+	Name       string
+	Netmask    string
+	IPAdresses []string
+}
+
+// Get the netmask for a given interface.  I used the ifconfig command for this.
+func getNetmask(interfaceName string) string {
+
+	netmaskStr := ""
+	args := []string{interfaceName}
+	output, err := exec.Command("ifconfig", args...).Output()
+	if err != nil {
+		// Don't fail, just return empty string
+		return netmaskStr
+	} else {
+		rows := strings.Split(string(output), "\n")
+		netmaskPos := strings.Index(rows[1], "netmask")
+		broadcastPos := strings.Index(rows[1], "broadcast")
+		if netmaskPos != -1 && broadcastPos != -1 {
+			netmaskStr = rows[1][netmaskPos+8 : broadcastPos]
+		}
+	}
+	return netmaskStr
+}
+
+// Get the IP address for a given interface.  There can be 0, 1 or 2 (e.g. IPv4 and IPv6)
+func getIPAddresses(iface net.Interface) []string {
+
+	var IPAdresses []string
+
+	addrs, err := iface.Addrs()
+	if err != nil {
+		return IPAdresses
+	}
+
+	for _, addr := range addrs {
+		IPAdresses = append(IPAdresses, "  "+addr.String())
+	}
+	return IPAdresses
+}
+
+// Get the IP address and net mask for each interface.
+func getInterfaceProperties(interfaces []net.Interface) []interfaceDetails {
+
+	var ifacesDetails []interfaceDetails
+
+	for _, iface := range interfaces {
+		netmask := getNetmask(iface.Name)
```

Comment:
You don't need this at all. There isn't one netmask per interface, it's per address, and you're already getting it when you get the IP addresses. The netmask is represented by the "/24" (or other number) that's already being printed at the end of each address.

The separate retrieval and output of netmasks can be removed.

FYI: "/24" is equivalent to "255.255.255.0". See also: https://www.iplocation.net/subnet-mask

Reply:
Ok, thanks Menno, I will fix it.